graylog2-web-interface/src/components/events/events/EventDetails.tsx

```diff
@@ -15,116 +15,37 @@
+    {children}
+  </DropdownButton>
+);
 
-    return eventDefinitionTypes.find((edt) => edt.type === event.event_definition_type);
-  }, [event, eventDefinitionTypes]);
+const EventDetails = ({ event, eventDefinitionContext }: Props) => {
+  const meta = useMemo(() => ({ context: { event_definitions: eventDefinitionContext } }), [eventDefinitionContext]);
 
   return (
-    <Row>
-      <Col md={6}>
-        <dl>
-          <dt>ID</dt>
-          <dd>{event.id}</dd>
-          <dt>Priority</dt>
-          <dd>
-            <PriorityName priority={event.priority} />
-          </dd>
-          <dt>Timestamp</dt>
-          <dd> <Timestamp dateTime={event.timestamp} />
-          </dd>
-          <dt>Event Definition</dt>
-          <dd>
-            <EventDefinitionLink event={event} eventDefinitionContext={eventDefinitionContext} />
-            &emsp;
-            ({(plugin && plugin.displayName) || event.event_definition_type})
-          </dd>
-          <dt>Remediation Steps</dt>
-          <dd>
-            {eventDefinitionContext?.remediation_steps ? (
-              <MarkdownPreview show
-                               withFullView
-                               noBorder
-                               noBackground
-                               value={eventDefinitionContext.remediation_steps} />
-            ) : (
-              <i>No remediation steps</i>
-            )}
-          </dd>
-          {!event.event_definition_type.startsWith('system-notifications') && (
-            <>
-              <dt>Actions</dt>
-              {event.replay_info && (
-              <dd>
-                <LinkToReplaySearch id={event.id} isEvent />
-              </dd>
-              )}
-              {pluggableActions}
-            </>
-          )}
-        </dl>
-      </Col>
-      <Col md={6}>
-        <dl>
-          {event.timerange_start && event.timerange_end && (
-            <>
-              <dt>Aggregation time range</dt>
-              <dd>
-                <Timestamp dateTime={event.timerange_start} />
-                &ensp;&mdash;&ensp;
-                <Timestamp dateTime={event.timerange_end} />
-              </dd>
-            </>
-          )}
-          <dt>Event Key</dt>
-          <dd>{event.key || 'No Key set for this Event.'}</dd>
-          <dt>Additional Fields</dt>
-          {isEmpty(event.fields)
-            ? <dd>No additional Fields added to this Event.</dd>
-            : <EventFields fields={event.fields} />}
-          <dt>Group-By Fields</dt>
-          {isEmpty(event.group_by_fields)
-            ? <dd>No group-by fields on this Event.</dd>
-            : <EventFields fields={event.group_by_fields} />}
-        </dl>
-      </Col>
-    </Row>
+    <MetaDataProvider<EventsAdditionalData> meta={meta}>
+      <EventDetailsTable attributesList={attributesList} event={event} meta={meta} />
+      <EventActions event={event} wrapper={ActionsWrapper} />
+    </MetaDataProvider>
```

Comment:
Do I understand it correctly that we are overwriting the meta data context, provided by the entity data table, here? So depending on where in the table you consume the context you get different data? Can't we just get rid of the context here and just provide the `eventDefinition` as a prop?

Reply:
@linuspahl I added Provider here because, in the EventDetailsTable component, we are using column renderers to render parameters in the same way as we render them on the Events Entity Data Table.  Some of the renderers need metadata. This [EventDetails.tsx](https://github.com/Graylog2/graylog2-server/pull/20831/files/9326ec1fd7fce39d2c528bb6d07c4f53ae81d566#diff-657844be72515f848a3d5b2a260f15bae2db440ff2341328e7cea48a93a5abcb) component is in use in Events widget. And events widget doesn't have any meta providers. So we don't override it, because the provider doesn't exist here. I can move this component to the widget section to avoid misconception